### DISCUSSION TITLE
Sync error when trying to update a one-to-many relationship

### DISCUSSION THREAD

**Discussion by Baspa** [laravel/nova-issues#3333] [Q&A]
- Laravel Version: 7.0
- Nova Version: 3.22
- PHP Version: 7.1.3

### Description:
I have a `Client` model that can be related to many `Team` models:
```
    public function teams()
    {
        return $this->hasMany(Team::class);
    }
```

The `Team` model can only be related to one `Client` model:
```
    public function client()
    {
        return $this->hasOne(Client::class, 'id', 'client_id');
    }
```
I want to be able to update this Client in Laravel Nova and add more relationships. For example: a team switches to another Client so I can select another Team in the update form and add it to the Client. I tried using both [Benjacho/belongs-to-many-field-nova](https://github.com/Benjacho/belongs-to-many-field-nova) and [dillingham/nova-attach-many](https://github.com/dillingham/nova-attach-many) packages to be able to edit this model and its relationship. However, no matter which package I use I constantly get the same error:

> Call to undefined method Illuminate\Database\Eloquent\Relations\HasMany::sync()

Because I get this error every time I assume this has something to do with Laravel Nova or the way I defined my relationships in Laravel.  

### Detailed steps to reproduce the issue on a fresh Nova installation:

1. Create a Client and a Team model in Laravel and Nova
2. Define a one-to-many relationship between those two models
3. Try to update the Client model

**Comment by crynobone** [comment]
`sync()` only used for attaching relation with pivot.
